openshift_tools/monitoring/zbxapi.py

```diff
@@ -0,0 +1,256 @@
+        'Dservice': ['get'],
+        'Event': ['acknowledge', 'get'],
+        'Graph': ['create', 'delete', 'get', 'update'],
+        'Graphitem': ['get'],
+        'Graphprototype': ['create', 'delete', 'get', 'update'],
+        'History': ['get'],
+        'Hostgroup': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'massadd', 'massremove', 'massupdate', 'update'],
+        'Hostinterface': ['create', 'delete', 'get', 'massadd', 'massremove', 'replacehostinterfaces', 'update'],
+        'Host': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'massadd', 'massremove', 'massupdate', 'update'],
+        'Hostprototype': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Httptest': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Iconmap': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Image': ['create', 'delete', 'get', 'update'],
+        'Item': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Itemprototype': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Maintenance': ['create', 'delete', 'get', 'update'],
+        'Map': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Mediatype': ['create', 'delete', 'get', 'update'],
+        'Proxy': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Screen': ['create', 'delete', 'get', 'update'],
+        'Screenitem': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'update', 'updatebyposition'],
+        'Script': ['create', 'delete', 'execute', 'get', 'getscriptsbyhosts', 'update'],
+        'Service': ['adddependencies', 'addtimes', 'create', 'delete', 'deletedependencies', 'deletetimes', 'get', 'getsla', 'isreadable', 'iswritable', 'update'],
+        'Template': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'massadd', 'massremove', 'massupdate', 'update'],
+        'Templatescreen': ['copy', 'create', 'delete', 'get', 'isreadable', 'iswritable', 'update'],
+        'Templatescreenitem': ['get'],
+        'Trigger': ['adddependencies', 'create', 'delete', 'deletedependencies', 'get', 'isreadable', 'iswritable', 'update'],
+        'Triggerprototype': ['create', 'delete', 'get', 'update'],
+        'User': ['addmedia', 'create', 'delete', 'deletemedia', 'get', 'isreadable', 'iswritable', 'login', 'logout', 'update', 'updatemedia', 'updateprofile'],
+        'Usergroup': ['create', 'delete', 'get', 'isreadable', 'iswritable', 'massadd', 'massupdate', 'update'],
+        'Usermacro': ['create', 'createglobal', 'delete', 'deleteglobal', 'get', 'update', 'updateglobal'],
+        'Usermedia': ['get'],
+    }
+
+    def __init__(self, data=None):
+        if not data:
+            data = {}
+        self.server = data.get('server', None)
+        self.username = data.get('username', None)
+        self.password = data.get('password', None)
```

Comment:
I'd prefer a NamedTuple called "ZabbixConnection" or something like that to doing a dictionary.

I hate passing in a single dictionary for all params, that makes me have to parse the function to find out what's supposed to be in the dictionary in the first place.

Also, it increases the likelihood of not finding typos.